staging/src/k8s.io/kube-aggregator/pkg/controllers/status/available_controller.go

```diff
@@ -184,8 +199,38 @@ func (c *AvailableConditionController) sync(key string) error {
 			return err
 		}
 	}
-
-	// TODO actually try to hit the discovery endpoint
+	// actually try to hit the discovery endpoint when it isn't local and when we're routing as a service.
+	// TODO figure out if the kube-proxy code handling has similar problems
+	if apiService.Spec.Service != nil && !c.enableAggregatorRouting && c.serviceResolver != nil {
```

Comment:
I believe the problem your pointing out exists either way. The aggregator has invalid endpoints and attempts to route the request itself to that bad endpoint then I assume that it will fail. If the aggregator routing is disabled then our request will be routed via kube-dns or a hardware load-balancer which is likely configured via a custom controller. In either case the set of ip addresses that they resolve to will also be based on the bad endpoints. So this will also cause the request to fail. In my opinion the only bad part of this is that this means you can get inconsistent results on the request, however that is true on either path.